### PR TITLE
doc: Automatically generate API documentation for the AuthProxyWorkload

### LINKED ISSUE #165: doc: Generate API documentation for the AuthProxyWorkload CRD
Add a new make target that creates generated API documentation for AuthProxyWorkload. This should
be run as part of `make generate`

### PR DESCRIPTION
**Pull request by hessjcg**
This will automatically generate API documentation for the AuthProxyWorkload CRD
and put it into the source code at `docs/api.md`. This uses the tool 
[crd-ref-docs](https://github.com/elastic/crd-ref-docs). The api docs are generated
automatically during `make generate` so that the docs commited to the source code 
are always in sync with go source. 

Fixes #165

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# API Reference
```
Comment (hessjcg) [inline docs/api.md:1]: This file is automatically generated by the crd-ref-docs tool

File: config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2022 Google LLC
```
Comment (hessjcg) [inline config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml:1]: This file is generated. Changes in this file reflect the changes to godoc in `authproxyworkload_types.go`

File: config/rbac/role.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2022 Google LLC
```
Comment (hessjcg) [inline config/rbac/role.yaml:1]: This file is generated.

File: config/webhook/manifests.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2022 Google LLC
```
Comment (hessjcg) [inline config/webhook/manifests.yaml:1]: This file is generated.

File: installer/cloud-sql-proxy-operator.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2022 Google LLC
```
Comment (hessjcg) [inline installer/cloud-sql-proxy-operator.yaml:1]: This file is generated. Changes in this file reflect the changes to godoc in `authproxyworkload_types.go`

File: internal/api/v1alpha1/authproxyworkload_types.go [hunk 6]
```diff
@@ -58,27 +58,40 @@ const (
 	ReasonUpToDate = "UpToDate"
 )
 
-// AuthProxyWorkloadSpec defines the desired state of AuthProxyWorkload
```
Comment (hessjcg) [inline internal/api/v1alpha1/authproxyworkload_types.go:61]: These are documentation changes intended to improve published api documentation.

File: tools/config-crd-ref-docs.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 Google LLC
```
Comment (hessjcg) [inline tools/config-crd-ref-docs.yaml:1]: This is the configuration file for our crd-ref-docs tool used to generate `docs/api.md`

**Review by hessjcg — COMMENTED** [review]
Some guidance for reviewers.

File: installer/cloud-sql-proxy-operator.yaml [hunk 8]
```diff
@@ -48,7 +48,7 @@ spec:
     - name: v1alpha1
       schema:
         openAPIV3Schema:
-          description: AuthProxyWorkload declares how a Cloud SQL Proxy container should be applied to a matching set of workloads, and shows the status of those proxy containers. This is the Schema for the authproxyworkloads API.
+          description: AuthProxyWorkload declares how a Cloud SQL Proxy container should be applied to a matching set of workloads, and shows the status of those proxy containers.
```
Comment (hessjcg) [inline installer/cloud-sql-proxy-operator.yaml:51]: All changes in this file are generated by controller-gen from the godoc in `authproxyworkload_types.go`

File: config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml [hunk 9]
```diff
@@ -30,7 +30,7 @@ spec:
     - name: v1alpha1
       schema:
         openAPIV3Schema:
-          description: AuthProxyWorkload declares how a Cloud SQL Proxy container should be applied to a matching set of workloads, and shows the status of those proxy containers. This is the Schema for the authproxyworkloads API.
+          description: AuthProxyWorkload declares how a Cloud SQL Proxy container should be applied to a matching set of workloads, and shows the status of those proxy containers.
```
Comment (hessjcg) [inline config/crd/bases/cloudsql.cloud.google.com_authproxyworkloads.yaml:33]: All changes in this file are generated by controller-gen from the godoc in `authproxyworkload_types.go`

**Review by enocom — APPROVED** [review]

**Pull request merged**